### DISCUSSION TITLE
Generating Terraform provider lock file in repositories with nested directory structure

### DISCUSSION THREAD

**Discussion by ntrianta** [renovatebot/renovate#14894] [Request Help]
Hello everybody,

We were trying to use renovate as a solution to keep our Terraform provider versions up to date. Our Terraform code repository follows a nested directory structure where each bottom-level directory is its own self-contained module and maintains its own providers. 

At first I was really happy to see that with minimal configuration Renovate was able to open a single MR with all the files that referenced a specific provider updated. Shortly after I realized that `terraform verify` was failing because the versions lock file (`.terraform.lock.hcl`) was not being generated. 

After digging through the documentation I found out that while this is supported as an experimental feature, it works only when the provider versions file lives in the root directory of the repository. 

Would you be so kind to please first of all verify for me that this is indeed the case (nested structure not supported) and I haven't misunderstood something? If yes, could you please tell me if you have plans to support this in the future?

If this is relevant to the discussion, I would like to add that we tried to run renovate self-hosted, as a gitlab runner. 

Thank you very much in advance! 😃 

Regards,
Nick

**Comment by rarkins** [comment]
@secustor could you comment?

**Comment by viceice** [comment]
please create a minimal reproduction, i know it works in subdir.

i think the versions file must be on same level as the lockfile.

**Comment by secustor** [comment]
The lock file has to be in the root of the Terraform files as described here in the [Terraform docs](https://www.terraform.io/language/files/dependency-lock#lock-file-location), but as @viceice has already mentioned this TF_ROOT has not to be the root of the repository.    
e.g. such a file structure is supported: 
```
/README.md
/terraform/main.tf
/terraform/.terraform.lock.hcl
/terraform/modules/
...
```

To better help you, a reproduction repo would be necessary.
